.github/workflows/ci.yaml

```diff
@@ -0,0 +1,68 @@
+# This workflow will install Python dependencies, run tests and lint with a matrix of OS and Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: CI
+
+on:
+  push:
+    branches:
+      - '*'
+      - '!master'
+  pull_request:
+    branches: [ master ]
+
+jobs:
```

Comment:
One thing we don't have currently here to allow for what @rafalp suggests is `scripts/check`, i.e. a "run linters only" script. We could run it in a separate task from `scripts/test`, so that we'd know earlier on which of checks or actual tests have failed.

One quite good example of this approach is HTTPX's [`.travis.yml`](https://github.com/encode/httpx/blob/master/.travis.yml). I also use this extensively in my projects, eg see [aiometer `azure-pipelines.yml`](https://github.com/florimondmanca/aiometer/blob/master/ci/azure-pipelines.yml).

Eventually the pipeline could probably look like this:

```yaml
jobs:
  check:
    steps:
    # Runs scripts/install, then scripts/check
  test:
    strategy: ...
    steps:
    # Runs scripts/install, then scripts/test
```

Reply:
yes, jobs are available as well, but i'm not sure the matrix {os|python-verion} can be reused at a higher level, so we can duplicate it actually save steps with installed environments as artifacts and pass it further.